app/models/post.rb

```diff
@@ -0,0 +1,17 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :comments
+  has_many :likes
+
+  def self.posts_counter(user)
+    where(author: user).count
+  end
+
+  def self.update_posts_counter(user)
+    User.select(:id).find(user.id).update(posts_counter: posts_counter(user))
+  end
+
```

Comment:
- [ ] This method is intended for an instance of a post, not the entire class. We don't need to define the method this way. Kindly remove the `self.` and the `(user)` and define this method as an instance method (suggestion below)

- [ ] **[OPTIONAL]** We can simply improve this method by making it shorter. We also don't need to comb through the entire users table to find the `author` when we already have access to the `author` through `post.author` (remember the post belongs to an `author`)

  ```
  def update_posts_counter
    author.increment!(:posts_counter)
  end
  ```